### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by Shinigami92**
highly complex shit XD

### CODE REVIEW

File: Assets/XXHash.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@
 using System;
 using System.IO;
 
-namespace xxHashSharp
+namespace RandomGenerator
```
Comment (Shinigami92) [inline Assets/XXHash.cs:34]: I generated a namespace for our random generators

File: Assets/Player.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ private void Awake()
 
 	private void Start()
 	{
-		Invoke(nameof(EnableGravity), 2f);
+		Invoke(nameof(EnableGravity), 10f);
```
Comment (Shinigami92) [inline Assets/Player.cs:26]: I have to solve that later in a different way 🤔

File: Assets/MeshFaceDirection.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+public enum MeshFaceDirection
+{
+	Front,
```
Comment (Shinigami92) [inline Assets/MeshFaceDirection.cs:3]: I read that in C# the naming convention for enum values is PascalCase

File: Assets/CacheOrigin.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+public enum CacheOrigin
```
Comment (Shinigami92) [inline Assets/CacheOrigin.cs:1]: This is not currently used, but can help later

File: Assets/Chunk.cs [hunk 5]
```diff
@@ -38,23 +31,53 @@ private void Start()
...
-				double noise = ChunkGenerator.NoiseGenerator.Octave(
-					numIterations: 3,
-					x: noiseX,
-					y: noiseZ,
-					persistence: 0.4,
-					scale: 0.1,
-					low: 0,
-					high: 16
-				);
+				if (i > 8)
```
Comment (Shinigami92) [inline Assets/Chunk.cs:72]: This is where the magic happens

File: Assets/Chunk.cs [hunk 6]
```diff
@@ -65,47 +88,47 @@ private void Start()
...
-
-	private void UpdateMesh()
+	private IEnumerator UpdateMesh()
 	{
 		List<CombineInstance> meshes = new List<CombineInstance>();
 
+		int i = 0;
 		foreach (KeyValuePair<Vector3Int, Block> item in data)
 		{
+			if (i > 64)
```
Comment (Shinigami92) [inline Assets/Chunk.cs:126]: This is where the magic happens

File: Assets/WorldGenerator.cs [hunk 7]
```diff
@@ -2,56 +2,61 @@
 using System.Collections.Generic;
 using System.Linq;
 using UnityEngine;
-using xxHashSharp;
+using RandomGenerator;
 
-public class ChunkGenerator : MonoBehaviour
+public class WorldGenerator : MonoBehaviour
```
Comment (Shinigami92) [inline Assets/WorldGenerator.cs:7]: Now that we also generate regions, not just chunks, this name fits better

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**